packages/core/src/internal/services/focus-trap-tracker.service.ts

```diff
@@ -1,84 +1,43 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { hasAttributeAndIsNotEmpty, setAttributes } from '../utils/dom.js';
 import { arrayTail } from '../utils/array.js';
+import { GlobalStateService } from './global.service.js';
+
+export const CDS_FOCUS_TRAP_DOCUMENT_ATTR = 'cds-focus-trap';
 
 /**
- * FocusTrapTracker is a static class that keeps track of the active element with focus trap,
+ * FocusTrapTrackerService is a static class that keeps track of the active element with focus trap,
  * in case there are multiple in a given page.
  */
-
-export const CDS_FOCUS_TRAP_ID_ATTR = 'focus-trap-id';
-export const CDS_FOCUS_TRAP_DOCUMENT_ATTR = 'cds-focus-trap-ids';
-
-export class FocusTrapTracker {
-  static getDocroot(): HTMLElement {
-    return document.documentElement as HTMLElement;
+export class FocusTrapTrackerService {
+  static getTrapElements(): { focusTrapId: string }[] {
+    return [...GlobalStateService.state.focusTrapRegistry];
   }
 
-  static getTrapIds(): string[] {
-    const docroot = this.getDocroot();
-
-    if (hasAttributeAndIsNotEmpty(docroot, CDS_FOCUS_TRAP_DOCUMENT_ATTR)) {
-      // the function in the conditional handles all nil references. zero chance of null making it through here.
-      const myAttribute = docroot.getAttribute(CDS_FOCUS_TRAP_DOCUMENT_ATTR) || '';
-      // TS forcing us to write an unreachable codepath. this is where monads would be useful.
-      return myAttribute === '' ? [] : myAttribute.split(' ');
-    } else {
-      return [];
-    }
+  static setTrapElements(elements: { focusTrapId: string }[]): void {
+    elements.length
+      ? document.documentElement.setAttribute(CDS_FOCUS_TRAP_DOCUMENT_ATTR, '')
```

Comment:
I don't think this will work in React. When I was investigating the modal backdrop-click and esc keypress not firing, this is where it would die. `documentElement` was not defined. Maybe you've done something here differently such that it works now. But I was moving towards `document.querySelector("html")` because that was coming back defined in React.

Reply:
I wish I could find a way to test this consistently in react.